app/models/spree/back_in_stock_notification.rb

```diff
@@ -34,6 +34,16 @@ def pending?
     email_sent_at.nil?
   end
 
+  def self.emails_of_ready_to_send(stock_location)
+    pending.in_stock(stock_location).pluck(:email)
+  end
+
+  def self.ready_to_send(email, stock_location)
+    pending
+      .where(email: email, stock_location: stock_location)
+      .in_stock(stock_location)
+  end
```

Comment:
Perhaps this could be written as a scope and go along with the scopes above.

```
  scope :ready_to_send, -> (email, stock_location) do
    pending
      .where(email: email, stock_location: stock_location)
      .in_stock(stock_location)
  end
```

Reply:
Thanks. Yes that makes sense. I also renamed the scope to `ready_to_send_by_email` to make it clear it is not all ready to send notifications.

I also moved `emails_of_ready_to_send` to a scope, but I'm not sure if this is a wrong use of a scope as it uses `pluck` instead of an `ActiveRecord` `select`.